s3.py

```diff
@@ -220,12 +220,12 @@ def cover_image_url(self, data_source, identifier, filename,
         parts = [identifier.type, identifier.identifier, filename]
         return root + self.key_join(parts)
 
-    def marc_file_url(self, library, lane):
+    def marc_file_url(self, library, lane, start_time=None, end_time=None):
         """The path to the hosted MARC file for the given library, lane,
         and date range."""
         bucket = self.get_bucket(self.MARC_BUCKET_KEY)
         root = self.marc_file_root(bucket, library)
-        parts = [lane.display_name]
+        parts = [start_time, end_time, lane.display_name]
```

Comment:
I'm not crazy about the way this creates URLS like None/None, as seen in the test below. It looks like there will be an end_time in every realistic case. So how about making end_time required and either doing /end_time/ or /start_time-end_time/ as one path segment?